Programowanie3/Assets/DamageOverTime.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DamageOverTime : MonoBehaviour
+{
+    private float damageTimer;
+    public float dmg;
+
+    private void OnTriggerEnter(Collider collision)
+    {
+        Debug.Log("kolizja");
+        Health hp = collision.gameObject.GetComponent<Health>();
+        StartCoroutine(DmgOverTime(hp, 6f));
```

Comment:
```suggestion
        Health hp = collision.GetComponent<Health>();
        if(hp != null)
        StartCoroutine(DmgOverTime(hp, 6f));
```
Tu przydałoby się sprawdzić, czy na tym obiekcie jest Health, bo inaczej może być błąd. I GetComponent możesz zrobić na komponencie, nie muszisz wyciągać gameObjectu